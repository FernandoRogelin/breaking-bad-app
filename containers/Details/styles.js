import styled from 'styled-components';

import ButtonComponent from '#/components/Button';
import ContentComponent from '#/components/Content';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
`;

export const Title = styled.h1`
  margin: 0;
  font-size: 1.5em;
`;

export const Content = styled(ContentComponent)`
  padding: 15px 0;
`;

export const Text = styled.p`
  margin: 0 0 0 10px;
`;

export const Info = styled.p`
  color: #8c9aa3;
  margin: 0 0 0 30px;

  &:not(:last-child) {
    margin-bottom: 10px;
  }
`;

export const Header = styled.header`
  display: flex;
  position: relative;
  margin-bottom: 26px;
  align-items: center;
`;

export const Button = styled(ButtonComponent)`
  left: -80px;
  padding: 1px 6px;
  position: absolute;

  @media only screen and (max-width: 440px) {
    left: -60px;
  }
`;
