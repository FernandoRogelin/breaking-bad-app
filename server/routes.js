const routes = require('next-routes');

module.exports = routes()
  .add('login')
  .add('episodes')
  .add('characters')
  .add('quotes');
