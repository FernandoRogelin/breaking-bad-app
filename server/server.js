const Koa = require('koa');
const Router = require('koa-router');
const helmet = require('koa-helmet');
const logger = require('koa-logger');
const bodyParcer = require('koa-bodyparser');
const nextjs = require('next');

const nextRoutes = require('./routes');
const port = process.env.PORT || 8080;
const app = nextjs({ dev: process.env.NODE_ENV !== 'production' });
const nextHandler = nextRoutes.getRequestHandler(app);

app
  .prepare()
  .then(() => {
    const server = new Koa();
    const router = new Router();

    server
      .use(logger())
      .use(bodyParcer())
      .use(helmet());

    router.get('/', ctx => {
      ctx.redirect('/login');
    });

    // Handle NextJS pages
    router.get('*', async ctx => {
      await nextHandler(ctx.req, ctx.res);

      ctx.status = 200;
      ctx.respond = false;
    });

    server.use(router.routes(), router.allowedMethods());

    server.listen(port, err => {
      if (err) console.log(err);
      console.log(`Server running in port http://localhost:${port}`);
    });
  })
  .catch(ex => {
    console.log(ex.stack);
    process.exit(1);
  });
