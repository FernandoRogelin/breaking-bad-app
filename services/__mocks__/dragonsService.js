export const getDragons = jest.fn().mockResolvedValue({
  status: 200,
  data: [
    {
      id: '54',
      createdAt: '2019-04-13T18:42:21.806Z',
      name: 'Ysera',
      type: 'Verde',
      histories: [],
    },
    {
      id: '56',
      createdAt: '2019-04-14T07:31:43.898Z',
      name: 'Rosamond',
      type: 'azure',
      histories: [],
    },
    {
      id: '57',
      createdAt: '2019-04-13T23:44:08.488Z',
      name: 'Leopoldo',
      type: 'white',
      histories: [],
    },
    {
      id: '61',
      createdAt: '2019-04-14T09:40:39.780Z',
      name: 'Sadie',
      type: 'orange',
      histories: [],
    },
    {
      id: '62',
      createdAt: '19/05/2018',
      name: 'Dinquedo',
      type: 'Amarelo',
      histories: [],
    },
  ],
});

export const deleteDragon = jest.fn().mockResolvedValue({
  status: 200,
});

export const getDetailsDragon = jest.fn().mockResolvedValue({
  status: 200,
  data: {
    id: '62',
    createdAt: '19/05/2018',
    name: 'Dinquedo',
    type: 'Amarelo',
    histories: [],
  },
});
