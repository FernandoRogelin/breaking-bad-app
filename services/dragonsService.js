import axios from 'axios';

export const urlBase =
  'http://5c4b2a47aa8ee500142b4887.mockapi.io/api/v1/dragon';

export const getDragons = () => axios.get(urlBase);

export const getDetailsDragon = id => axios.get(`${urlBase}/${id}`);

export const createDragon = ({ name, type, createdAt }) =>
  axios.post(urlBase, {
    name,
    type,
    createdAt,
  });

export const deleteDragon = id => axios.delete(`${urlBase}/${id}`);

export const editDragon = (id, { name, type, createdAt }) =>
  axios.put(`${urlBase}/${id}`, {
    name,
    type,
    createdAt,
  });
