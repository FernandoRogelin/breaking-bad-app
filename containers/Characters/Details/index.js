import React, { useState, useEffect } from 'react';

import { format } from 'date-fns';
import { getCharacterDetails } from '#/services/breakingBadService';

import {
  Wrapper,
  Title,
  Content,
  Text,
  Info,
  Header,
  Button,
  Items,
} from './styles';

export default function Details({ id, setStateDetailsPage }) {
  const [characterDetails, setCharacterDetails] = useState();
  const [fetched, setFetched] = useState(false);

  async function handleCharacterDetails() {
    try {
      const { data } = await getCharacterDetails(id);

      setFetched(true);
      setCharacterDetails(data);
    } catch (e) {
      console.log(e);
    }
  }

  useEffect(() => {
    if (!fetched) {
      handleCharacterDetails();
    }
  }, [fetched]);

  return (
    <Wrapper>
      <Header>
        <Button onClick={setStateDetailsPage}>Return</Button>
        <Title>Character Details</Title>
      </Header>
      {characterDetails && (
        <Content>
          {characterDetails[0].name && (
            <Items>
              <Text>Name:</Text>
              <Info>{characterDetails[0].name}</Info>
            </Items>
          )}
          {characterDetails[0].portrayed && (
            <Items>
              <Text>Portrayed:</Text>
              <Info>{characterDetails[0].portrayed}</Info>
            </Items>
          )}
          {characterDetails[0].nickname && (
            <Items>
              <Text>Nickname:</Text>
              <Info>{characterDetails[0].nickname}</Info>
            </Items>
          )}
          {characterDetails[0].birthday && (
            <Items>
              <Text>Birthday:</Text>
              <Info>{format(characterDetails[0].birthday, 'MM/DD/YYYY')}</Info>
            </Items>
          )}
          {characterDetails[0].status && (
            <Items>
              <Text>Status:</Text>
              <Info>{characterDetails[0].status}</Info>
            </Items>
          )}
        </Content>
      )}
    </Wrapper>
  );
}
