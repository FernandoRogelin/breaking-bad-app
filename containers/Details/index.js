import React, { useState, useEffect } from 'react';

import { Router } from '#/server/routes';
import { getDetailsDragon } from '#/services/dragonsService';

import { Wrapper, Title, Content, Text, Info, Header, Button } from './styles';

export default function Details() {
  const [details, setDetails] = useState();
  const [called, setCalled] = useState(false);

  async function handleDragon() {
    try {
      const { data } = await getDetailsDragon(Router.query.id);
      setDetails(data);
      setCalled(true);
    } catch (e) {
      console.log(e);
    }
  }

  useEffect(() => {
    if (!called) {
      handleDragon();
    }
  }, [called]);

  return (
    <Wrapper>
      <Header>
        <Button onClick={() => Router.pushRoute('timeline')}>Voltar</Button>
        <Title>Detalhes do dragão</Title>
      </Header>
      {details && (
        <Content>
          <Text>Nome:</Text>
          {details && <Info>{details.name}</Info>}
          <Text>Tipo:</Text>
          {details && <Info>{details.type}</Info>}
          <Text>Dia de criação:</Text>
          {details && <Info>{details.createdAt}</Info>}
        </Content>
      )}
    </Wrapper>
  );
}
