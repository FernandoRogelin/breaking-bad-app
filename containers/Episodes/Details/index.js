import React, { useState, useEffect } from 'react';

import { format } from 'date-fns';
import { getEpisodeDetails } from '#/services/breakingBadService';

import {
  Wrapper,
  Title,
  Content,
  Text,
  Info,
  Header,
  Button,
  Items,
} from './styles';

export default function Details({ id, setStateDetailsPage }) {
  const [episodeDetails, setEpisodeDetails] = useState();
  const [fetched, setFetched] = useState(false);

  async function handleCharacterDetails() {
    try {
      const { data } = await getEpisodeDetails(id);

      setFetched(true);
      setEpisodeDetails(data);
    } catch (e) {
      console.log(e);
    }
  }

  useEffect(() => {
    if (!fetched) {
      handleCharacterDetails();
    }
  }, [fetched]);

  return (
    <Wrapper>
      <Header>
        <Button
          data-testid="return-episodes-page"
          onClick={setStateDetailsPage}
        >
          Return
        </Button>
        <Title>Episodes Details</Title>
      </Header>
      {episodeDetails && (
        <Content>
          {episodeDetails[0].title && (
            <Items>
              <Text>Title:</Text>
              <Info>{episodeDetails[0].title}</Info>
            </Items>
          )}
          {episodeDetails[0].season && (
            <Items>
              <Text>Season:</Text>
              <Info>{episodeDetails[0].season}</Info>
            </Items>
          )}
          {episodeDetails[0].characters && (
            <Items>
              <Text>Characters:</Text>
              <Info>{episodeDetails[0].characters}</Info>
            </Items>
          )}
          {episodeDetails[0].air_date && (
            <Items>
              <Text>Air Date:</Text>
              <Info>{format(episodeDetails[0].air_date, 'DD/MM/YYYY')}</Info>
            </Items>
          )}
        </Content>
      )}
    </Wrapper>
  );
}
