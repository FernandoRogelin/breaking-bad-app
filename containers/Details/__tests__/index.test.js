import React from 'react';
import { render, wait } from 'react-testing-library';
import { Router } from '#/server/routes';

import { getDetailsDragon } from '#/services/dragonsService';
import Details from '..';

Router.pushRoute = jest.fn();

const mockedRouter = { query: { id: () => {} } };
Router.router = mockedRouter;

jest.mock('#/services/dragonsService');

describe('Details container', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Details snapshot', async () => {
    const { container } = render(<Details />);

    await wait(() =>
      expect(getDetailsDragon).toHaveBeenCalledWith(mockedRouter.query.id)
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
