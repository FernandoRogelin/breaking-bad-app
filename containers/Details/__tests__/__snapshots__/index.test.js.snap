// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details container Details snapshot 1`] = `
<div
  class="styles__Wrapper-ff0pn0-0 eBWitS"
>
  <header
    class="styles__Header-ff0pn0-5 cBeasn"
  >
    <button
      class="Button-n3cyj4-0 styles__Button-ff0pn0-6 eejYfC"
    >
      Voltar
    </button>
    <h1
      class="styles__Title-ff0pn0-1 dwayCI"
    >
      Detalhes do dragão
    </h1>
  </header>
  <div
    class="Content-vb879o-0 styles__Content-ff0pn0-2 feoYOp"
  >
    <p
      class="styles__Text-ff0pn0-3 kDptvo"
    >
      Nome:
    </p>
    <p
      class="styles__Info-ff0pn0-4 kqkGTk"
    >
      Dinquedo
    </p>
    <p
      class="styles__Text-ff0pn0-3 kDptvo"
    >
      Tipo:
    </p>
    <p
      class="styles__Info-ff0pn0-4 kqkGTk"
    >
      Amarelo
    </p>
    <p
      class="styles__Text-ff0pn0-3 kDptvo"
    >
      Dia de criação:
    </p>
    <p
      class="styles__Info-ff0pn0-4 kqkGTk"
    >
      19/05/2018
    </p>
  </div>
</div>
`;
